clientv3/retry.go

```diff
@@ -54,14 +63,50 @@ func (c *Client) newRetryWrapper() retryRpcFunc {
 	}
 }
 
+func (c *Client) newRdonlyRetryWrapper() retryRpcFunc {
+	return func(rpcCtx context.Context, f rpcFunc) error {
+		for {
+			err := f(rpcCtx)
+			if err == nil {
+				return nil
+			}
+
+			// always stop retry on etcd errors other than invalid auth token
+			eErr := rpctypes.Error(err)
+			if etcdErr, ok := eErr.(rpctypes.EtcdError); ok {
+				if etcdErr.Code() == rpctypes.ErrInvalidAuthToken.(rpctypes.EtcdError).Code() {
+					gterr := c.getToken(rpcCtx)
+					if gterr != nil {
+						return err // return the original error for simplicity
+					}
+					continue
+
+				}
+				return err
+			}
+
+			return err
+		}
+	}
+}
+
 type retryKVClient struct {
 	pb.KVClient
-	retryf retryRpcFunc
+	retryf       retryRpcFunc
+	rdonlyRetryf retryRpcFunc
```

Comment:
`RetryClientKV()` would returning what's there now + the auth retries seems OK. I don't think there needs to be a special configuration option.

The main thing is I think the kv client wrapper internals should distinguish between wrapping for only writes/state changes (at-most-once rpcs) and wrapping  both reads and writes (either at-most-once or at-least-once rpcs). What's exposed to the client can be a composition of those wrappers so long what's returned implements the pb.KVClient interface.